components/arm/eva/eva.go

```diff
@@ -392,10 +405,43 @@ func (e *eva) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error)
 }
 
 func (e *eva) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	return e.MoveToJointPositions(ctx, e.model.ProtobufFromInput(goal), nil)
+	positionDegs := e.model.ProtobufFromInput(goal)
+	if err := checkDesiredJointPositions(ctx, e, positionDegs.Values); err != nil {
+		return err
+	}
+	return e.MoveToJointPositions(ctx, positionDegs, nil)
 }
 
 // Model returns the kinematics model of the eva arm, also has all Frame information.
 func Model(name string) (referenceframe.Model, error) {
 	return referenceframe.UnmarshalModelJSON(evamodeljson, name)
 }
+
+// checkDesiredJointPositions validates that the desired joint positions either bring the joint back
+// in bounds or do not move the joint more out of bounds.
+func checkDesiredJointPositions(ctx context.Context, a arm.Arm, desiredJoints []float64) error {
```

Comment:
Ack- this is way worse. Sorry, I didn't think through the request enough. Could you revert this and make it public (and only existing once) again?